runtime/program_lifecycle.md

```diff
@@ -98,3 +111,95 @@ In other words, you can use `addEventListener` to register multiple `"load"` or
 `"unload"` event handlers, but only the last defined `onload`, `onbeforeunload`,
 `onunload` event handlers will be executed. It is preferable to use
 `addEventListener` when possible for this reason.
+
+## `beforeunload` Example
+
+```js
+// beforeunload.js
+let count = 0;
+
+console.log(count);
+
+globalThis.addEventListener("beforeunload", (e) => {
+  console.log("About to exit...");
+  if (count < 4) {
+    e.preventDefault();
+    console.log("Scheduling more work...");
+    setTimeout(() => {
+      console.log(count);
+    }, 100);
+  }
+
+  count++;
+});
+
+globalThis.addEventListener("unload", (e) => {
+  console.log("Exiting");
+});
+
+count++;
+console.log(count);
+
+setTimeout(() => {
+  count++;
+  console.log(count);
+}, 100);
+```
+
+Running this program will print:
+
+```sh
+$ deno run beforeunload.js
+0
+1
+2
+About to exit...
+Scheduling more work...
+3
+About to exit...
+Scheduling more work...
+4
+About to exit...
+Exiting
+```
+
+This has allowed us to polyfill `process.on("beforeExit")` in the Node
+compatibility layer.
+
+## `unhandledrejection` event Example:
+
+This release adds support for the unhandledrejection event. This event is fired
+when a promise that has no rejection handler is rejected, ie. a promise that has
+no .catch() handler or a second argument to .then().
+
+```js
+// unhandledrejection.js
+globalThis.addEventListener("unhandledrejection", (e) => {
+  console.log("unhandled rejection at:", e.promise, "reason:", e.reason);
+  e.preventDefault();
+});
+
+function Foo() {
+  this.bar = Promise.reject(new Error("bar not available"));
+}
+
+new Foo();
+Promise.reject();
+```
+
+Running this program will print:
+
+```sh
+$ deno run unhandledrejection.js
+unhandled rejection at: Promise {
+  <rejected> Error: bar not available
+    at new Foo (file:///dev/unhandled_rejection.js:7:29)
+    at file:///dev/unhandled_rejection.js:10:1
+} reason: Error: bar not available
+    at new Foo (file:///dev/unhandled_rejection.js:7:29)
+    at file:///dev/unhandled_rejection.js:10:1
+unhandled rejection at: Promise { <rejected> undefined } reason: undefined
+```
+
+This API will allow us to polyfill `process.on("unhandledRejection")` in the
+Node compatibility layer in future releases.
```

Comment:
The support of `process.on("unhandledRejection")` in Node compat layer is already available (https://github.com/denoland/deno_std/pull/2460). I think this sentence should be dropped.